src/IO/S3/Client.cpp

```diff
@@ -749,42 +759,40 @@ RequestResult Client::processRequestResult(RequestResult && outcome) const
     return RequestResult(error);
 }
 
-void Client::sleepAfterNetworkError(Aws::Client::AWSError<Aws::Client::CoreErrors> error, Int64 attempt_no) const
+void Client::updateNextTimeToRetryAfterRetryableError(Aws::Client::AWSError<Aws::Client::CoreErrors> error, Int64 attempt_no) const
 {
-    auto sleep_ms = client_configuration.retryStrategy->CalculateDelayBeforeNextRetry(error, attempt_no);
-    if (!client_configuration.s3_slow_all_threads_after_network_error)
-    {
-        LOG_WARNING(log, "Request failed, now waiting {} ms before attempting again", sleep_ms);
-        sleepForMilliseconds(sleep_ms);
```

Comment:
I think this is incorrect because it changes the behavior of the old versions with `s3_slow_all_threads_after_network_error` disabled.